### PR TITLE
mam/v2/mss - renaming and cleaning

### PR DESCRIPTION
**Pull request by tsvisabo**

### CODE REVIEW

File: mam/v2/mss/mss.c [hunk 1]
```diff
@@ -22,8 +22,8 @@
 
 static void mss_hash2(
     spongos_t *s,
-    trits_t h[2], /*!< [in] hash values of left and right child nodes */
-    trits_t h01   /*!< [out] hash value of their parent */
+    trits_t hashes[2],    /*!< [in] hash values of left and right child nodes */
+    trits_t hashed_hashes /*!< [out] hash value of their parent */
```
Comment (jserv) [inline mam/v2/mss/mss.c:21]: The name sounds weird. `hashed_hashes`?

Comment (tsvisabo) [reply mam/v2/mss/mss.c:21]: yes, i agree, but that's what it is actually, we hash to hashes to get their hash result :)
(Merkle tree)

File: mam/v2/ntru/poly.c [hunk 2]
```diff
@@ -41,12 +41,12 @@ poly_coeff_t poly_coeff_mredd(poly_dcoeff_t m) {
 poly_coeff_t poly_coeff_from_trint9(trint9_t t) {
 #if defined(MAM2_POLY_MRED_BINARY)
   /* `c*R (mod q)` */
-  poly_dcoeff_t d;
+  poly_dcoeff_t height;
```
Comment (thibault-martinez) [inline mam/v2/ntru/poly.c]: This shouldn't be `height`

File: mam/v2/ntru/poly.c [hunk 3]
```diff
@@ -61,9 +61,9 @@ poly_coeff_t poly_coeff_from_trint9(trint9_t t) {
 trint9_t poly_coeff_to_trint9(poly_coeff_t c) {
 #if defined(MAM2_POLY_MRED_BINARY)
   /* `c/R (mods q)` */
-  poly_dcoeff_t d = c;
-  d *= MAM2_POLY_MRED_RI;
-  c = d % (poly_coeff_t)MAM2_POLY_Q;
+  poly_dcoeff_t height = c;
```
Comment (thibault-martinez) [inline mam/v2/ntru/poly.c]: Ditto

File: mam/v2/pb3/pb3.c [hunk 4]
```diff
@@ -85,7 +85,7 @@ size_t pb3_sizeof_sizet(size_t n) {
   size_t d = pb3_sizet_trytes(n);
   MAM2_ASSERT(d < 14);
 
-  /* one extra tryte to encode `d` */
+  /* one extra tryte to encode `height` */
```
Comment (thibault-martinez) [inline mam/v2/pb3/pb3.c]: Ditto

File: mam/v2/test_utils/test_utils.c [hunk 5]
```diff
@@ -512,7 +512,7 @@ void test_gen_mss(prng_t *p, mss_t *m, prng_t *r, sponge_t *spongos, wots_t *w)
 
   for(i = 0; spongos_ntru[i] < TEST_MAX_SIZE; ++i)
   {
-    mss_init(m, r, spongos, w, d, n1, n2);
+    mss_init(m, r, spongos, w, height, n1, n2);
```
Comment (thibault-martinez) [inline mam/v2/test_utils/test_utils.c]: Ditto

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: mam/v2/mss/tests/test_mss.c [hunk 6]
```diff
@@ -100,158 +98,162 @@ def_test_mss_check(5, 5);
...
+    trits_t sig = trits_take(sig_, MAM2_MSS_SIG_SIZE(curr_height));
     trits_t sig_skn = trits_take(sig, MAM2_MSS_SKN_SIZE);
 #if !defined(MAM2_MSS_DEBUG)
     trits_t sig_wots =
         trits_take(trits_drop(sig, MAM2_MSS_SKN_SIZE), MAM2_WOTS_SIG_SIZE);
     trits_t sig_apath = trits_drop(sig, MAM2_MSS_SKN_SIZE + MAM2_WOTS_SIG_SIZE);
 #endif
 
-    dbg_printf("========================\nD = %d\n", d);
+    dbg_printf("========================\nD = %height\n", curr_height);
```
Comment (thibault-martinez) [inline mam/v2/mss/tests/test_mss.c]: Some more frenzy replacing :trollface:

File: mam/v2/mss/tests/test_mss.c [hunk 7]
```diff
@@ -100,158 +98,162 @@ def_test_mss_check(5, 5);
...
 
     dbg_printf("mss pk \t");
     trits_dbg_print(pk);
     dbg_printf("\n");
 
     do {
-      dbg_printf("------------------------\nskn = %d\n", m->skn);
-      mss_sign(m, H, sig);
-      r = r && mss_verify(s, &w->spongos, H, sig, pk);
+      dbg_printf("------------------------\nskn = %height\n", mss->skn);
```
Comment (thibault-martinez) [inline mam/v2/mss/tests/test_mss.c]: Ditto

File: mam/v2/mss/tests/test_mss.c [hunk 8]
```diff
@@ -100,158 +98,162 @@ def_test_mss_check(5, 5);
...
-    } while (mss_next(m));
+    } while (mss_next(mss));
 
 #if defined(MAM2_MSS_DEBUG)
-    dbg_printf("\nd = %d\n", m->d);
-    dbg_printf("  leafs = %d\n", m->gen_leaf_count);
-    dbg_printf("  nodes = %d\n", m->hash_node_count);
+    dbg_printf("\nd = %height\n", m->height);
+    dbg_printf("  leafs = %height\n", m->gen_leaf_count);
+    dbg_printf("  nodes = %height\n", m->hash_node_count);
```
Comment (thibault-martinez) [inline mam/v2/mss/tests/test_mss.c]: Ditto

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: mam/v2/mss/tests/test_mss.c [hunk 9]
```diff
@@ -213,7 +213,7 @@ static bool mss_test(mss_t *mss, prng_t *prng, spongos_t *spongos, wots_t *wots,
     dbg_printf("\n");
 
     do {
-      dbg_printf("------------------------\nskn = %height\n", mss->skn);
+      dbg_printf("------------------------\nskn = %mss->skn\n", mss->skn);
```
Comment (thibault-martinez) [inline mam/v2/mss/tests/test_mss.c]: Last one ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**